### PR TITLE
Fix `test_rolling_agg_aggregate` for `pandas` 2.0 compatibility

### PR DESCRIPTION
**Pull request by j-bennet**
Fixes an error in upstream:

```
dask/dataframe/tests/test_rolling.py::test_rolling_agg_aggregate: TypeError: _mean_dispatcher() got an unexpected keyword argument 'args'
```

Pandas 2.0 fixes a bug where args passed to applied aggs were ignored. We need the same fix.

Xref https://github.com/pandas-dev/pandas/pull/50863.
Xref https://github.com/dask/dask/issues/9736.

- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/rolling.py [hunk 1]
```diff
@@ -496,7 +496,9 @@ def _has_single_partition(self):
     @staticmethod
     def pandas_rolling_method(df, rolling_kwargs, name, *args, **kwargs):
         rolling = df.rolling(**rolling_kwargs)
-        return getattr(rolling, name)(*args, **kwargs)
+        pd_args = kwargs.get("args", None) or ()
+        pd_kwargs = kwargs.get("kwargs", None) or {}
+        return getattr(rolling, name)(*args, *pd_args, **pd_kwargs)
```
Comment (jrbourbeau) [inline dask/dataframe/rolling.py:501]: What do you think about something like 

```diff
diff --git a/dask/dataframe/rolling.py b/dask/dataframe/rolling.py
index 5636312e8..29031f52f 100644
--- a/dask/dataframe/rolling.py
+++ b/dask/dataframe/rolling.py
@@ -612,10 +612,8 @@ class Rolling:
         )

     @derived_from(pd_Rolling)
-    def aggregate(self, func, args=(), kwargs=None, **kwds):
-        if kwargs is None:
-            kwargs = {}
-        return self._call_method("agg", func, args=args, kwargs=kwargs, **kwds)
+    def aggregate(self, func, *args, **kwargs):
+        return self._call_method("agg", func, *args, **kwargs)

     agg = aggregate
```

instead? If we can just pass args and kwargs through at the `aggregate`, then we may not need the special handling here in `pandas_rolling_method` (though I could very well be missing something)

Comment (j-bennet) [reply dask/dataframe/rolling.py]: Lovely. ❤️ I applied your suggestion, looks so much better. I thought there was some reason for this clunky method signature with `kwargs` and `kwds`, but all the tests in `dask/dataframe/tests/test_rolling.py` are passing, so it looks like there is no reason after all.

**Comment by j-bennet** [comment]
Test that fails in CI passes locally:

```
FAILED dask/tests/test_distributed.py::test_blockwise_dataframe_io[True-None-hdf] - AssertionError: DataFrame are different
```

**Review by jrbourbeau — APPROVED** [review]
This looks great, thanks @j-bennet

**Pull request merged**